### PR TITLE
v22.2 Add autoResizeEnabled for TextArea

### PR DESCRIPTION
**Pull request by H-Frederick**
Modified demos:
- jQuery
- Angular
- AngularJS
- React
- Vue
- ASP.NET Core
- ASP.NET MVC

### CODE REVIEW

**Review by flagmanAndrew — DISMISSED** [review]

File: JSDemos/Demos/TextArea/Overview/Angular/app/app.component.ts [hunk 1]
```diff
@@ -38,16 +42,18 @@ export class AppComponent {
...
-      this.value = this.service.getContent();
-      this.maxLength = null;
-    }
+    let str = this.service.getContent();
+    this.value = e.value ? str.substring(0, 100) : str;
+    this.maxLength = e.value ? 100 : null;
+  }
+
+  onAutoResizeChanged(e) {
+    this.autoResizeEnabled = e.value;
```
Comment (Zedwag) [inline JSDemos/Demos/TextArea/Overview/Angular/app/app.component.ts]: as far as we use 2way binding, there is no need to save `autoResizeEnabled ` value here, it should be alredy changed

File: JSDemos/Demos/TextArea/Overview/Vue/App.vue [hunk 2]
```diff
@@ -65,17 +73,19 @@ export default {
...
     onCheckboxValueChanged(e) {
-      if (e.value) {
-        this.value = service.getContent().substring(0, 100);
-        this.maxLength = 100;
-      } else {
-        this.value = service.getContent();
-        this.maxLength = null;
-      }
+      const str = service.getContent();
+      this.value = e.value ? str.getContent().substring(0, 100) : str;
```
Comment (Zedwag) [inline JSDemos/Demos/TextArea/Overview/Vue/App.vue]:
```suggestion
      this.value = e.value ? str.substring(0, 100) : str;
```

File: JSDemos/Demos/TextArea/Overview/Vue/App.vue [hunk 3]
```diff
@@ -65,17 +73,19 @@ export default {
...
+      this.maxLength = e.value ? 100 : null;
+    },
+    onAutoResizeChanged(e) {
+      this.autoResizeEnabled = e.value;
```
Comment (Zedwag) [inline JSDemos/Demos/TextArea/Overview/Vue/App.vue]: just as in angular, there is no need to save value here

File: JSDemos/Demos/TextArea/Overview/jQuery/index.js [hunk 4]
```diff
@@ -7,17 +7,22 @@ $(() => {
...
     value: false,
     onValueChanged(data) {
-      if (data.value) {
-        exampleTextArea.option('value', (exampleTextArea.option('value')).substring(0, 100));
-        exampleTextArea.option('maxLength', 100);
-      } else {
-        exampleTextArea.option('maxLength', null);
-        exampleTextArea.option('value', longText);
-      }
+      const str = data.value ? (exampleTextArea.option('value')).substring(0, 100) : longText;
```
Comment (Zedwag) [inline JSDemos/Demos/TextArea/Overview/jQuery/index.js]:
```suggestion
      const str = data.value ? exampleTextArea.option('value').substring(0, 100) : longText;
```

File: MVCDemos/Views/TextArea/Overview.cshtml [hunk 5]
```diff
@@ -56,13 +63,16 @@
...
         var exampleTextArea = $("#example-textarea").dxTextArea("instance");
-        if(data.value) {
-            exampleTextArea.option("value", (exampleTextArea.option("value")).substring(0, 100));
-            exampleTextArea.option("maxLength", 100);
-        } else {
-            exampleTextArea.option("maxLength", null);
-            exampleTextArea.option("value", text);
-        }
+
+        exampleTextArea.option("value", data.value ? (exampleTextArea.option("value")).substring(0, 100) : null);
```
Comment (Zedwag) [inline MVCDemos/Views/TextArea/Overview.cshtml]:
```suggestion
        exampleTextArea.option("value", data.value ? exampleTextArea.option("value").substring(0, 100) : null);
```

File: NetCoreDemos/Views/TextArea/Overview.cshtml [hunk 6]
```diff
@@ -56,13 +63,16 @@
...
         var exampleTextArea = $("#example-textarea").dxTextArea("instance");
-        if(data.value) {
-            exampleTextArea.option("value", (exampleTextArea.option("value")).substring(0, 100));
-            exampleTextArea.option("maxLength", 100);
-        } else {
-            exampleTextArea.option("maxLength", null);
-            exampleTextArea.option("value", text);
-        }
+
+        exampleTextArea.option("value", data.value ? (exampleTextArea.option("value")).substring(0, 100) : null);
```
Comment (Zedwag) [inline NetCoreDemos/Views/TextArea/Overview.cshtml]:
```suggestion
        exampleTextArea.option("value", data.value ? exampleTextArea.option("value").substring(0, 100) : null);
```

File: JSDemos/Demos/TextArea/Overview/AngularJS/index.html [hunk 7]
```diff
@@ -21,6 +21,9 @@
         <div class="dx-field">
           <div dx-check-box="checkBoxOptions"></div>
         </div>
+        <div class="dx-field">
```
Comment (Zedwag) [inline JSDemos/Demos/TextArea/Overview/AngularJS/index.html:24]: just for future, we ussually do not update angular js demos anymore

**Review by Zedwag — CHANGES_REQUESTED** [review]

**Review by Zedwag — APPROVED** [review]

**Review by flagmanAndrew — APPROVED** [review]

**Pull request merged**